tests/Domain/Auth/Controllers/AccessControllerTest.php

```diff
@@ -0,0 +1,199 @@
+
+        $this->accessController = new AccessController(
+            $this->auth0,
+            $this->integrationRepository,
+            ['admin@publiq.be'],
+            $this->logger
+        );
+    }
+
+    public function testInvalidTokenFails(): void
+    {
+        $this->logger->expects($this->once())
+            ->method('info')
+            ->with('Requested IntegrationAccess for integration 9469af87-3455-4853-a156-dbbc24965a26');
+
+        $this->logger->expects($this->once())
+            ->method('warning')
+            ->with('Invalid token', ['exception' => 'The JWT string must contain two dots']);
+
+        $this->integrationRepository->expects($this->never())
+            ->method('getById');
+
+        $response = $this->accessController->handle('invalid token', '9469af87-3455-4853-a156-dbbc24965a26');
+
+        $this->assertEquals(400, $response->getStatusCode());
+        $this->assertEquals(
+            ['exception' => 'The JWT string must contain two dots'],
+            Json::decodeAssociatively((string) $response->getContent())
+        );
+    }
+
+    public function testTokenWithMissingEmailFails(): void
+    {
+        $this->logger->expects($this->once())
+            ->method('info')
+            ->with('Requested IntegrationAccess for integration 9469af87-3455-4853-a156-dbbc24965a26');
+
+        $this->logger->expects($this->once())
+            ->method('warning')
+            ->with('No email in token');
+
+        $this->integrationRepository->expects($this->never())
+            ->method('getById');
+
+        $response = $this->accessController->handle(
+            'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c',
+            '9469af87-3455-4853-a156-dbbc24965a26'
+        );
+
+        $this->assertEquals(400, $response->getStatusCode());
+        $this->assertEquals(
+            ['exception' => 'No email in token'],
+            Json::decodeAssociatively((string) $response->getContent())
+        );
+    }
+
+    public function testAdminHasAccess(): void
+    {
+        $this->logger->expects($this->exactly(2))
+            ->method('info')
+            ->with(
+                $this->logicalOr(
+                    'Requested IntegrationAccess for integration 9469af87-3455-4853-a156-dbbc24965a26',
+                    'Admin access for admin@publiq.be'
+                )
+            );
+
+        $this->integrationRepository->expects($this->never())
+            ->method('getById');
+
+        $response = $this->accessController->handle(
+            'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiZW1haWwiOiJhZG1pbkBwdWJsaXEuYmUiLCJpYXQiOjE1MTYyMzkwMjJ9.Haq7nxdrHejchsa22_JFUevoYF_vpEFBjW9iSdN2CF4',
+            '9469af87-3455-4853-a156-dbbc24965a26'
+        );
+
+        $this->assertEquals(200, $response->getStatusCode());
+    }
```

Comment:
Weird formatting

Reply:
The token is very long, could it be that? This call has 2 parameters, the first is the token and the second is the integration ID.

Reply:
It looks better now 👍🏻 

Probably due to github